orm/src/main/java/org/hibernate/search/batchindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -24,6 +24,7 @@
 import org.hibernate.engine.spi.SessionFactoryImplementor;
 import org.hibernate.engine.spi.SessionImplementor;
 import org.hibernate.engine.transaction.jta.platform.spi.JtaPlatform;
+import org.hibernate.internal.CriteriaImpl;
```

Comment:
no we're not absolutely sure, but this class has been there for ~10 years.

As soon as the 5.7 final is out we'll set up CI jobs, to discuss with the ORM team any breaking changes.

Getting rid of the warning is quite important, both from performance and user confusion perspective, and I couldn't find a better solution.

One alternative would be to patch ORM to introduce an explicit SPI level contract for this, but then you'd need to bump the requirement to > 5.2.8 ... I prefer this solution.